File: Java/opentraceability/src/main/java/opentraceability/models/common/AnonymizedAddress.java
package opentraceability.models.common;

import opentraceability.utility.Country;

public class AnonymizedAddress {
    public String zipCode = null;
    public Country country = null;

    public boolean equals(Object obj) {
        if (obj == null) {
            return false;
        }

        if (!(obj instanceof AnonymizedAddress)) {
            return false;
        }

        return this.equals((AnonymizedAddress) obj);
    }

    public boolean equals(AnonymizedAddress other) {
        if (other == null) {
            return false;
        }

        return (this.zipCode == other.zipCode && this.country == other.country);
    }

    public int hashCode() {
        return this.zipCode.hashCode() + this.country.hashCode();
    }
}

File: Kotlin/OpenTraceability/src/main/kotlin/opentraceability/models/common/anonymizedaddress.kt
package opentraceability.models.common

import opentraceability.utility.Country

open class AnonymizedAddress {
    open var zipCode: String? = null
    open var country: Country? = null


    override fun equals(obj: Any?): Boolean {

        if (obj == null)
        {
            return false
        }

        if (obj !is AnonymizedAddress)
        {
            return false
        }

        return this.equals(obj as AnonymizedAddress)
    }

    fun equals(other: AnonymizedAddress?): Boolean {

        if (other != null){
            return false
        }

        return (this.zipCode == other?.zipCode && this.country == other?.country);
    }

    override fun hashCode(): Int {
        return this.zipCode.hashCode() + this.country.hashCode()
    }

}
